### PR TITLE
Use #!/bin/bash for script with bash-specific commands

### PR DESCRIPTION
**Pull request by roehling**
The test/Resources/push.sh calls pushd/popd, which is not a POSIX shell feature but a Bash extension.

### CODE REVIEW

File: tests/resources/push.sh [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
```
Comment (ethomson) [inline tests/resources/push.sh:1]: Makes sense - but should we:

```suggestion
#!/usr/bin/env bash
```

Comment (roehling) [reply tests/resources/push.sh:1]: I don't think that's necessary. The `#!/usr/bin/env foo` idiom was invented for things like Python scripts, which might be run from a virtual environment, so you cannot hard-code the interpreter path. Bash has no such use-case and `/bin/bash` is not going to go away any time soon, so why pay the cost for an additional `exec("/usr/bin/env")` syscall?

Comment (ethomson) [reply tests/resources/push.sh:1]: This isn't in a tight loop or a place where any syscall is expensive. But the performance question needs to take a backseat to correctness and usability -- where is `bash` on a Solaris system?

Comment (roehling) [reply tests/resources/push.sh:1]: It's in `/bin/bash` AFAIK, but everything before Solaris 11 has a stone-age bash which is not even POSIX compliant.

Comment (neithernut) [reply tests/resources/push.sh:1]: Friends of mine in the NixOS-Community would likely point out that `bash` would need to be resolved by `/usr/bin/env` there too. IIRC their `/bin` only contains `sh` (and another link to/copy of `env`) but that may well have changed in the mean-time.

Comment (ethomson) [reply tests/resources/push.sh:1]: Right. The other option, of course, is that we get rid of the bashisms entirely which is also good.

**Comment by ethomson** [comment]
Thanks for the fix!

**Pull request merged**